### ISSUE TITLE
.set stored_query_result command returns 400 - Bad Request

### ISSUE THREAD

**Issue by MSDkuznetsov** [grafana/azure-data-explorer-datasource#820]
**What happened**:

I wanted to use the https://learn.microsoft.com/en-us/azure/data-explorer/kusto/management/set-stored-query-result-command to store the state.

According to documentation, to use this command "You must have [Database Viewer](https://learn.microsoft.com/en-us/azure/data-explorer/kusto/management/access-control/role-based-access-control) permissions to run these commands." - which is the case, I have confirmed that Grafana account has the appropriate permission, I have also tried to use my own principal with the same results.

![image](https://github.com/grafana/azure-data-explorer-datasource/assets/136453859/8a402720-63fc-4671-ac18-75cd071fb207)

**What you expected to happen**:

- To be able to store the state when dashboard is loaded (as a caching mechanism) or at least have some clarity about the restrictions if there are any, so that appropriate error message is shown - "not allowed to be executed, due to policy" something like that.

**How to reproduce it (as minimally and precisely as possible)**:

1. Go to any dashboard
2. Add new panel that relies on Azure Data Explorer as data source
3. Type in the following kql statement:

```kql
.set-or-replace stored_query_result Numbers3 <| range X from 1 to 10 step 1
```

4. Load the query
5. Failure will be shown

**Anything else we need to know?**:

btw reading stored query results, works just fine:

```kql
.show stored_query_results
```

![image](https://github.com/grafana/azure-data-explorer-datasource/assets/136453859/196895e8-9d77-4843-a53a-da691f55570b)

**Environment**:

- Grafana version: Azure Managed Grafana - Grafana v10.3.4 (5bde27379c)

**Comment by aangelisc** [comment]
Hi @MSDkuznetsov, thank you for the report. I can confirm I've replicated the behaviour in Grafana. However, I also see issues when executing this query in Azure directly:

![Image](https://github.com/grafana/azure-data-explorer-datasource/assets/15019026/fdc112d6-bf23-4aaa-9520-6e1e65151977)

Are you able to confirm if this is an Azure issue?

**Comment by MSDkuznetsov** [comment]
I see no issues from my side, works as expected. Based on the error it seems to me like it's some other error related to connection. 
![image](https://github.com/grafana/azure-data-explorer-datasource/assets/136453859/19e2c0aa-4dae-41ef-8c29-497c821d319c)


> Hi @MSDkuznetsov, thank you for the report. I can confirm I've replicated the behaviour in Grafana. However, I also see issues when executing this query in Azure directly:
> 
> ![Image](https://private-user-images.githubusercontent.com/15019026/314488944-fdc112d6-bf23-4aaa-9520-6e1e65151977.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTA5MzYwNDksIm5iZiI6MTcxMDkzNTc0OSwicGF0aCI6Ii8xNTAxOTAyNi8zMTQ0ODg5NDQtZmRjMTEyZDYtYmYyMy00YWFhLTk1MjAtNmUxZTY1MTUxOTc3LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAzMjAlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMzIwVDExNTU0OVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTdhNWFkYzk4ZTk1NWVkNjBkMTdkZjU2YjA0ZjcwNGIzNzk0YTg5ZWNhNDViYjBhNTkzNzVhNWRkYzJhYWQ2NDQmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.LQ4gsEBUBr_C_5D2xU_nHBvuhF_gB3fpXeFf9n_SoCk)
> 
> Are you able to confirm if this is an Azure issue?

**Comment by aangelisc** [comment]
Hi @MSDkuznetsov,

The actual underlying issue is that these queries aren't supported by the endpoint Grafana uses. The true error is:

```
Bad request: Control commands (starting with a dot '.') cannot be served from the query endpoint unless they are .show control commands.
```

I've added a task to improve the error messages so this is made clearer to the user.

**Comment by MSDkuznetsov** [comment]
> Hi @MSDkuznetsov,
> 
> The actual underlying issue is that these queries aren't supported by the endpoint Grafana uses. The true error is:
> 
> ```
> Bad request: Control commands (starting with a dot '.') cannot be served from the query endpoint unless they are .show control commands.
> ```
> 
> I've added a task to improve the error messages so this is made clearer to the user.

Thanks, that makes sense !

**Comment by aangelisc** [comment]
Fixed by #822
